src/pipelines/worldpop/worldpop_population.py

```diff
@@ -0,0 +1,125 @@
+from lib.net import download
+from lib.pipeline import DataSource
+from lib.cast import age_group
+
+
+def _aggregate_population(data: Series) -> Dict[str, int]:
+    """
+    WorldPop data is "double stacked" by breaking down by age *and* sex, whereas we only want
+    a breakdown of age *or* sex. This function converts the columns:
+    `[
+        m_0,
+        m_5,
+        ...
+        f_0,
+        f_5,
+        ...
+    ]`
+
+    Into:
+    `[
+        population,
+        population_male,
+        population_female,
+        population_age_00_09,
+        population_age_10_19,
+        ...
+    ]`
+    """
+    age_bucket_size = 10
+    age_bucket_count = 10
+    age_bucket_pairs = [
+        (i * age_bucket_size, (i + 1) * age_bucket_size - 1) for i in range(age_bucket_count)
+    ]
+
+    aggregated_values = {
+        "key": data["key"],
+        "population": 0,
+        "population_male": 0,
+        "population_female": 0,
+        **{f"population_age_{lo:02d}_{hi:02d}": 0 for lo, hi in age_bucket_pairs},
+    }
+    for col, val in data.iteritems():
+        # Skip over the key item
+        if col == "key":
+            continue
+
+        # Total population is the sum of all populations
+        aggregated_values["population"] += val
+
+        # Get age/sex info from column name
+        sex, age = col.split("_", 2)
+
+        # Add male and female populations separately
+        if sex == "m":
+            aggregated_values["population_male"] += val
+        elif sex == "f":
+            aggregated_values["population_female"] += val
+        else:
+            raise ValueError(f"Unexpected sex label encountered: {sex}")
+
+        # Go over all the age buckets and add them separately
+        # Since the WorldPop buckets are [0, 1-5, 5-9, 10-14, ...] we can just use the lower
+        # range of the bucket as the age value to convert into our normalized buckets which are
+        # [0-9, 10-19, 20-29, ...]
+        age_bucket = age_group(int(age))
+
+        # Make sure that the age buckets all follow the pattern \d\d_\d\d
+        age_bucket = "_".join([f"{int(age):02d}" for age in age_bucket.split("-", 2)])
+
+        aggregated_values[f"population_age_{age_bucket}"] += val
+
+    return aggregated_values
+
+
+class WorldPopPopulationDataSource(DataSource):
+    """
+    Retrieves demographics information from WorldPop for all items in metadata.csv. The original
+    data source is https://www.worldpop.org/project/categories?id=8 but the data is pre-processed
+    using Google Earth Engine: https://code.earthengine.google.com/f885dd559364ed8918324da355c1ee0e
+    """
+
+    def parse(self, sources: List[str], aux: Dict[str, DataFrame], **parse_opts) -> DataFrame:
+
+        # In this script, we don't have any sources since the Worldpop data is precomputed using
+        # Earth Engine: https://code.earthengine.google.com/f885dd559364ed8918324da355c1ee0e
+
+        # Compute the estimated total male population
```

Comment:
Why do you call out 'male' here specifically?  We're computing total male, total female, and total for each age bucket, correct?

Reply:
Good point, fixed.